app/src/main/java/emu/skyline/input/InputHandler.kt

```diff
@@ -175,6 +265,44 @@ class InputHandler(private val inputManager : InputManager, private val preferen
         return false
     }
 
+    override fun onAccuracyChanged(sensor : Sensor?, accuracy : Int) {}
+
+    /**
+     * This handles translating any [SensorEvent]s to a [GuestEvent] that is passed into libskyline
+     */
+    override fun onSensorChanged(event : SensorEvent) {
+        when (event.sensor.type) {
+            Sensor.TYPE_ACCELEROMETER -> {
+                motionSensor.accelerometer[0] = motionAcelOrientation[0] * event.values[1] / SensorManager.GRAVITY_EARTH
+                motionSensor.accelerometer[1] = motionAcelOrientation[1] * event.values[0] / SensorManager.GRAVITY_EARTH
+                motionSensor.accelerometer[2] = motionAcelOrientation[2] * event.values[2] / SensorManager.GRAVITY_EARTH
+            }
+
+            Sensor.TYPE_GYROSCOPE -> {
+                // Investigate why sensor value is off by 12x
+                motionSensor.gyroscope[0] = motionGyroOrientation[0] * event.values[1] / 12.0f
+                motionSensor.gyroscope[1] = motionGyroOrientation[1] * event.values[0] / 12.0f
+                motionSensor.gyroscope[2] = motionGyroOrientation[2] * event.values[2] / 12.0f
+            }
+
+            else -> {}
+        }
+
+        // Only update state on accelerometer data
+        if (event.sensor.type != Sensor.TYPE_ACCELEROMETER)
+            return
+
+        motionSensor.deltaTimestamp = event.timestamp.toULong() - motionSensor.timestamp
+        motionSensor.timestamp = event.timestamp.toULong()
+
+        motionDataBufferLeft.clear()
+        motionDataBufferRight.clear()
+        motionDataBufferHandheld.clear()
+        setMotionState(0, 0, motionSensor.writeToByteBuffer(motionDataBufferLeft))
+        setMotionState(0, 1, motionSensor.writeToByteBuffer(motionDataBufferRight))
+        setMotionState(0, 2, motionSensor.writeToByteBuffer(motionDataBufferHandheld))
```

Comment:
Since `motionDataBuffer` is a throwaway buffer, you can avoid having three different ones by clearing them before each call, like this:
```suggestion
        motionDataBuffer.clear()
        setMotionState(0, 0, motionSensor.writeToByteBuffer(motionDataBuffer))
        motionDataBuffer.clear()
        setMotionState(0, 1, motionSensor.writeToByteBuffer(motionDataBuffer))
        motionDataBuffer.clear()
        setMotionState(0, 2, motionSensor.writeToByteBuffer(motionDataBuffer))
```